src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -63,36 +63,53 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     addTermRating(request);
     response.setContentType("text/html; charset=UTF-8");
     response.setCharacterEncoding("UTF-8");
-    response.sendRedirect("/index.html");
   }
 
   public void addTermRating(HttpServletRequest request) throws IOException {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    String termFeedback = request.getParameter("term-input");
-    Long termRating = Long.parseLong(request.getParameter("rating-term"));
-    Long workHours = Long.parseLong(request.getParameter("hoursOfWork"));
-    Long difficulty = Long.parseLong(request.getParameter("difficulty"));
-    String professorFeedback = request.getParameter("prof-input");
-    Long professorRating = Long.parseLong(request.getParameter("rating-professor"));
+
+    StringBuilder jb = new StringBuilder();
+    String line = null;
+    try {
+      BufferedReader reader = request.getReader();
+      while ((line = reader.readLine()) != null) jb.append(line);
+    } catch (Exception exception) {
+      /*report an error*/
```

Comment:
Right now this will silently fail, do you want to throw an error or something here?

Reply:
FYI typically an empty catch statement is an anti-pattern, as it hides errors from you at runtime. Ideally either let it fail in some way (throwing or not catching), or tell in a comment exactly why this isn't actually an error and it's fine to silently catch it.